AWSKinesis/AWSFirehoseRecorder.m

```diff
@@ -177,8 +177,9 @@ - (AWSTask *)submitRecordsForStream:(NSString *)streamName
     return [[self.firehose putRecordBatch:putRecordBatchInput] continueWithBlock:^id(AWSTask *task) {
         if (task.error) {
             AWSDDLogError(@"Error: [%@]", task.error);
-            if ([task.error.domain isEqualToString:NSURLErrorDomain]) {
+            if (task.error) {
```

Comment:
The check for NSURLErrorDomain was due to a bug where the call went to an infinite loop. We should be setting the stop variable only when the client goes offline. Removing the check would make the process stop on first error (even if its transient and should not be stopped)

Reply:
The error that is thrown in my test is actually `AWSCognitoIdentityErrorDomain`, so we might need to at least also catch that and stop the task early, because that will not recover when online again.